app/controllers/smart_proxies_controller.rb

```diff
@@ -45,6 +45,13 @@ def tftp_server
     requested_data(:tftp_server)
   end
 
+  def puppet_environments
+    envs = @smart_proxy.public_send(:environment_stats)
```

Comment:
`requested_data` renders a json response, here I want to render a partial with that data.